src/internal/connector/graph/middleware.go

```diff
@@ -259,6 +259,18 @@ func (mw RetryMiddleware) retryRequest(
 		case <-timer.C:
 		}
 
+		// we have to reset the original body reader for each retry, or else the graph
+		// compressor will produce a 0 length body following an error response such
```

Comment:
msgraph documentation seems to suggest that the compression handler should be added before retry handler to avoid this exact problem. Any reason we are not doing it this way in our code?

`This handler MUST be inserted BEFORE the [Retry handler](https://microsoftgraph.github.io/msgraph-sdk-design/middleware/RetryHanlder.md) to avoid compressing the request payload multiple times over.` 

https://microsoftgraph.github.io/msgraph-sdk-design/middleware/CompressionHandler.html

Reply:
Well that's _very interesting_, because we took this ordering from [the kiota http client factory code](https://github.com/microsoft/kiota-http-go/blob/main/kiota_client_factory.go#L78).

Considering that this mimics the graph client itself, I think I'm comfortable with the ordering.

Reply:
Good point. Might be worthwhile raising an issue with msgraph folks here https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/middleware/CompressionHandler.md